src/main/java/spoon/reflect/factory/CodeFactory.java

```diff
@@ -139,7 +139,7 @@ public CodeFactory(Factory factory) {
 		fieldReference.setDeclaringType(type);
 
 		CtFieldRead<Class<T>> fieldRead = factory.Core().createFieldRead();
-		fieldRead.setType(classType);
+		fieldRead.setType(factory.Core().clone(classType));
```

Comment:
@GerardPaligot it's not the matter of equality. Any _object_ cannot appear in ast more than once. Otherwise very subtle bugs with element.replace() appear.